gitnamehere/point-of-sale-group-project
# POS - Finish implementing the ordering process

**Ticket:** **[ATS-51: POS - Finish implementing the ordering process](https://cs375-pos-group-project.atlassian.net/browse/ATS-51?atlOrigin=eyJpIjoiYjgzYmJhMDljODFiNGZjOWI2MThiODc0ZDY2YmYyNjgiLCJwIjoiaiJ9)](https://cs375-pos-group-project.atlassian.net/browse/ATS-52?atlOrigin=eyJpIjoiYjA2YzRjNmJlNzFmNDcwOGFhY2IyZGY1YzIxMDNkYzMiLCJwIjoiaiJ9)**

- Added filtering for orders by date
- Added filtering for orders by item name
- Added number of individual items ordered and total profit made across orders with item
- Added overview on sales (total number of orders, overall profit, total discounts, total tips)
- Added a few more order details to sale report (discounts, tips, and items in order)
- Added item.csv to docs

Testing:
- `npm run start:dev`
- Place order(s) at `http://localhost:3000/pos/ordering/`
- Pay for order(s) at `http://localhost:3000/pos/payment/?id={whatever}`
- Verify total number of orders made is accurate
- Verify total profit is accurate (only accounts for orders that have been paid)
- Verify total discounts is accurate (only accounts for orders that have been paid)
- Verify total tips is accurate (only accounts for orders that have been paid)
- Verify filtering by date is functional (may place orders at different dates by changing your PC's time - don't forget to change it back!)
- Verify filtering by item name is functional (invalid names should display nothing)
